src/templates/dso_api/dynamic_api/api.html

```diff
@@ -63,8 +67,458 @@
+    window.onload = onPageLoad;
+  }
+
+  function onPageLoad() {
+    let page = PAGEURL.searchParams.get("page");
+    if(page) {
+      setPageLinks(page);
+    }
+    getData(PAGEURL).catch(exceptionHandler).then(parseData);
+  }
+
+  function getPageUrl(page) {
+    // Get the URL of another page of the same endpoint
+    let url = new URL(PAGEURL);
+    url.searchParams.set("page", page);
+    return url;
+  };
+
+  function setPageLinks(page) {
+    page = Number(page);
+    if(isNaN(page) || page<1){
+      return
+    }
+    isPaginated = true;
+    let pageLinksEl = document.getElementById("page-links");
+    let links = pageLinksEl.getElementsByTagName("li");
+    links = [...Array(links.length).keys()].map(
+      i => {
+      return {
+        "li":links.item(i), 
+        "a": links.item(i).getElementsByTagName("a")[0]
+      }
+    })
+    
+    // Set forward page link, but leave disabled for now.
+    links[6]["a"].setAttribute("href", getPageUrl(page+1));
+
+    // Set self and previous links
+    for (let i = 0; page-i && i<4; i++) {
+      let el = links[5-i];
+      el["a"].setAttribute("href", getPageUrl(page-i));
+      el["a"].innerHTML = page-i;
+      el["li"].classList.remove("hidden");
+    }
+    
+    if(page > 1) {
+      // Set back link
+      links[0]["a"].setAttribute("href", getPageUrl(page-1));
+      links[0]["li"].classList.remove("disabled");
+
+      if(page > 4) {
+        // Set page 1 link
+        links[1]["a"].setAttribute("href", getPageUrl(1));
+        links[1]["li"].classList.remove("hidden");
+      }
+      if(page > 5){
+        // Set ... link
+        links[2]["a"].innerHTML = "...";
+        links[2]["li"].classList.add("disabled");
+      }
+    }
+
+    // Show the links
+    let containerEl = document.getElementById("page-container");
+    containerEl.classList.add("active");
+  }
+
+  function getData(url) {
+    return new Promise(function(callback, err) {
+      let http = new XMLHttpRequest();
+      http.open("GET", url, true);
+      http.setRequestHeader("Accept", "application/hal+json");
+      http.send();
+      http.onreadystatechange = function () {
+        if (this.readyState == 2){
+            parseHeaders(this);
+        }
+        if (this.readyState == 4){
+          let result = JSON.parse(this.responseText);
+          try {   
+            callback(result);
+          } catch (error) {
+              err("Error: JSON corrupt");
+          }
+        }
+      }; 
+    }); 
+  }
+
+  function parseHeaders(response) {
+    // Set header fields
+    let headersElement = document.getElementById("response-headers");
+    headersElement.innerHTML = "<b>HTTP " + response.status + " " + response.statusText + "</b>\n";
+    
+    for(let header of HEADERS) {
+      headerValue = response.getResponseHeader(header);
+      if (headerValue){
+        let headerline =  "<b>" + header + ':</b> <span class="lit">' + response.getResponseHeader(header) + "</span>\n";
```

Comment:
Ik zou hier lekker template literals (backticks) gebruiken. Wel de HTML escapen.

Reply:
Kan er HTML in de response headers voorkomen?

Reply:
Ik zou het er niet op gokken. Straks zetten wij een `Next:` pagination link in de response header die query-string parameters overneemt. Altijd escapen dus :-)